docs/docs/guides/api-album-sync.md

```diff
@@ -0,0 +1,123 @@
+# API Album Sync (Python Script)
+
+This is an example of a python script for syncing an album to a local folder. This was used for a digital photoframe so the displayed photos could be managed from the immich web or app UI.
+
+The script is copied below in it's current form. A repository is hosted [here](https://git.orenit.solutions/open/immichalbumpull).
+
+:::danger
+This guide uses a generated API key. This key gives the same access to your immich instance as the user it is attached to, be careful how the config file is stored and transferred.
+:::
+
+### Prerequisites
+
+- Python 3.7+
+- [requests library](https://pypi.org/project/requests/)
+
+### Installing
+
+Copy the contents of 'pull.py' (shown below) to your chosen location or clone the repository:
+
+```bash
+git clone https://git.orenit.solutions/open/immichalbumpull
+```
+
+Edit or create the 'config.ini' file in the same directory as the script with the necessary details:
+
+```ini title='config.ini'
+[immich]
+# URL of target immich instance
+url = https://photo.example.com
+# API key from Account Settings -> API Keys
+apikey = xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+# Full local path to target directory
+destination = /home/photo/photos
+# immich album name
+album = Photoframe
+```
+
+### Usage
+
+Run the script directly:
+
+```bash
+./pull.py
+```
+
+Or from cron (every 5 minutes):
+
+```bash
+*/5 * * * * /usr/bin/python /home/user/immichalbumpull/pull.py
+```
+
+### Python Script
+
+```python title='pull.py'
+#!/usr/bin/env python
+
+import requests
+import configparser
+import os
+import shutil
+
+# Read config file
+config = configparser.ConfigParser()
+config.read('config.ini')
+
+url = config['immich']['url']
+apikey = config['immich']['apikey']
+photodir = config['immich']['destination']
+albumname = config['immich']['album']
+
+headers = {
+    'Accept': 'application/json',
+    'x-api-key': apikey
+}
+
+# Set up the directory for the downloaded images
+os.makedirs(photodir, exist_ok=True)
+
+# Get the list of albums from the API
+response = requests.get(url + "/api/album", headers=headers)
+
+# Parse the JSON response
+data = response.json()
+
+# Find the chosen album id
+for item in data:
+    if item['albumName'] == albumname:
+        albumid = item['id']
+
+# Get the list of photos from the API using the albumid
+response = requests.get(url + "/api/album/" + albumid, headers=headers)
+
+# Parse the JSON response and extract the URLs of the images
+data = response.json()
+image_urls = [image for image in data['assets']]
+
+# Download each image from the URL and save it to the directory
+headers = {
+    'Accept': 'application/octet-stream',
+    'x-api-key': apikey
+}
+
+photolist = []
+
+for id in image_urls:
+    url = "https://photo.eisner.uk/api/download/asset/" + str(id['id'])
```

Comment:
This still uses your domain. Also it looks like this assumes all files are jpg.

Reply:
Thanks, I'll fix both. The hardcoded jpg is just something that comes from it starting as a basic script for my use case but I can improve it, especially with helpful critique.

Reply:
I think d5331ead8d72f2e9009693cf888010e5b58e2094 both shows my hacky inexperience and fixes the hardcoded url and extension. The only way I could see to get it was another call to the api for the asset info.

Reply:
Oh nice. Yeah, maybe we should send it with a content-disposition header which could provide a filename in the download response itself.